### PR TITLE
sp-sandbox: move the sandbox module of `sp-core` into `sp-sandbox`

### PR DESCRIPTION
**Pull request by koushiro**
Signed-off-by: koushiro <koushiro.cqx@gmail.com>

- Move `sandbox` module of `sp-core` into the `env` module of sp-sandbox
- Reorder some imports and dependencies (by plugin)

### CODE REVIEW

**Comment by bkchr** [comment]
Can you please give some more explanation why you have done this change?

**Comment by koushiro** [comment]
Actually `sp_core::sandbox` module is only used for `sp-sandbox` and `sc-executor`.
Moving it to `sp-sandbox` seems more intuitive to most people, I think.

**Comment by koushiro** [comment]
@bkchr  what do you think?

**Comment by koushiro** [comment]
@bkchr what do you think?

File: client/executor/common/src/sandbox/wasmi_backend.rs [hunk 1]
```diff
@@ -18,16 +18,17 @@
...
-use sp_core::sandbox::HostError;
-use sp_wasm_interface::{FunctionContext, Pointer, ReturnValue, Value, WordSize};
 use std::rc::Rc;
 
 use wasmi::{
 	memory_units::Pages, ImportResolver, MemoryInstance, Module, ModuleInstance, RuntimeArgs,
 	RuntimeValue, Trap, TrapKind,
 };
 
+use codec::{Decode, Encode};
```
Comment (bkchr) [inline client/executor/common/src/sandbox/wasmi_backend.rs:28]: What is the reasoning behind moving these lines?

File: client/executor/wasmi/src/lib.rs [hunk 2]
```diff
@@ -18,27 +18,29 @@
...
 use sc_executor_common::{
 	error::{Error, MessageWithBacktrace, WasmError},
 	runtime_blob::{DataSegmentsSnapshot, RuntimeBlob},
 	sandbox,
 	util::MemoryTransfer,
 	wasm_runtime::{InvokeMethod, WasmInstance, WasmModule},
 };
-use sp_core::sandbox as sandbox_primitives;
 use sp_runtime_interface::unpack_ptr_and_len;
+use sp_sandbox::env;
```
Comment (bkchr) [inline client/executor/wasmi/src/lib.rs:40]:
```suggestion
use sp_sandbox::env as sandbox_env;
```

Or something like that. Just `env` is not that expressive.

File: client/executor/wasmtime/src/host.rs [hunk 3]
```diff
@@ -19,18 +19,20 @@
...
+
+use codec::{Decode, Encode};
 use sc_allocator::FreeingBumpHeapAllocator;
 use sc_executor_common::{
 	error::Result,
 	sandbox::{self, SupervisorFuncIndex},
 	util::MemoryTransfer,
 };
-use sp_core::sandbox as sandbox_primitives;
+use sp_sandbox::env;
```
Comment (bkchr) [inline client/executor/wasmtime/src/host.rs:32]: Same here for the renaming.

File: primitives/sandbox/src/embedded_executor.rs [hunk 4]
```diff
@@ -193,13 +200,18 @@ impl<T> ImportResolver for EnvironmentDefinitionBuilder<T> {
...
 		})?;
 		let memory = match *externval {
 			ExternVal::Memory(ref m) => m,
 			_ => {
-				debug!(target: TARGET, "Export {}:{} is not a memory", module_name, field_name);
+				log::debug!(
+					target: TARGET,
+					"Export {}:{} is not a memory",
+					module_name,
+					field_name
```
Comment (bkchr) [inline primitives/sandbox/src/embedded_executor.rs:213]:
```suggestion
					field_name,
```

File: primitives/sandbox/src/embedded_executor.rs [hunk 5]
```diff
@@ -162,13 +164,18 @@ impl<T> ImportResolver for EnvironmentDefinitionBuilder<T> {
...
 		})?;
 		let host_func_idx = match *externval {
 			ExternVal::HostFunc(ref idx) => idx,
 			_ => {
-				debug!(target: TARGET, "Export {}:{} is not a host func", module_name, field_name);
+				log::debug!(
+					target: TARGET,
+					"Export {}:{} is not a host func",
+					module_name,
+					field_name
```
Comment (bkchr) [inline primitives/sandbox/src/embedded_executor.rs:177]:
```suggestion
					field_name,
```

File: client/executor/common/src/sandbox.rs [hunk 6]
```diff
@@ -22,25 +22,25 @@
 
 #[cfg(feature = "wasmer-sandbox")]
 mod wasmer_backend;
-
 mod wasmi_backend;
 
+use std::{collections::HashMap, rc::Rc};
+
+use codec::Decode;
+use sp_sandbox::env;
```
Comment (bkchr) [inline client/executor/common/src/sandbox.rs:30]: Same renaming.

**Review by bkchr — APPROVED** [review]

**Comment by pepyakin** [comment]
TBH, I am not entirely convinced that this change is justified. Perhaps @athei may have a different opinion, but I personally wouldn't merge it since this IMO only adds churn.

**Comment by pepyakin** [comment]
Yes, it does. On the second thought, I think we may as well let it in.

**Review by athei — APPROVED** [review]
Makes sense to move it there from my point of view.

**Comment by athei** [comment]
Can you merge master?

**Pull request merged**